### PR TITLE
Add some extra tests to kavadist 

### PR DESCRIPTION
**Pull request by DracoLi**
A number of new tests added to the kavadist module

### CODE REVIEW

File: x/kavadist/genesis_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+					End:       time.Date(2021, 2, 1, 1, 1, 1, 1, time.UTC),
+					Inflation: sdk.OneDec(),
+				},
+			},
+		},
+		tmtime.Canonical(time.Unix(1, 0)),
+	)
+
+	kavadist.InitGenesis(suite.Ctx, suite.Keeper, suite.AccountKeeper, state)
+	suite.Equal(state.Params, suite.Keeper.GetParams(suite.Ctx))
```
[the remark this answers is not in the document]
Comment (rhuairahrighairidh) [reply x/kavadist/genesis_test.go:58]: I'm fine with either. I think there's use cases for each.

File: app/test_common.go [hunk 2]
```diff
@@ -158,6 +159,22 @@ func (tApp TestApp) FundAccount(ctx sdk.Context, addr sdk.AccAddress, amounts sd
 	return tApp.bankKeeper.SendCoinsFromModuleToAccount(ctx, minttypes.ModuleName, addr, amounts)
 }
 
+// NewQueryServerTestHelper creates a new QueryServiceTestHelper that wraps
```
[the remark this answers is not in the document]
Comment (DracoLi) [reply app/test_common.go:162]: vscode does seem to support multi-line comments. i'm not sure about godoc though so I can update this.

[the remark this answers is not in the document]
Comment (DracoLi) [reply app/test_common.go]: https://github.com/Kava-Labs/kava/pull/1038/commits/fcb3eeeaa4000417c99c491119d13b34bf56a967

File: x/kavadist/genesis_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+					Inflation: sdk.OneDec(),
+				},
+			},
+		},
+		tmtime.Canonical(time.Unix(1, 0)),
+	)
+
+	kavadist.InitGenesis(suite.Ctx, suite.Keeper, suite.AccountKeeper, state)
+	suite.Require().Equal(state.Params, suite.Keeper.GetParams(suite.Ctx))
+
```
Comment (rhuairahrighairidh) [inline x/kavadist/genesis_test.go:59]: nit: I'd maybe not use the `DefaultPreviousBlockTime` in `state` and instead check a normal time is imported and exported as expected. During mainnet upgrades, the genesis state `PreviousBlockTime` will never be the default.

For context, using `DefaultPreviousBlockTime` in the genesis state basically means the previous block time is unknown and should start from the next available block time. This is really just a convenience feature for starting chains from scratch.

File: x/kavadist/keeper/keeper_test.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func TestKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(keeperTestSuite))
+}
+
+func (suite *keeperTestSuite) TestGetPreviousBlock_NoPreviousBlock() {
+	blockTime, found := suite.Keeper.GetPreviousBlockTime(suite.Ctx)
+	suite.Require().False(found)
+	suite.Require().Equal(blockTime, types.DefaultPreviousBlockTime)
```
Comment (rhuairahrighairidh) [inline x/kavadist/keeper/keeper_test.go:28]: nit: It doesn't look like the anything depends on `DefaultPreviousBlockTime` being returned when the time is not found in the store. I'd probably remove the check and align `keeper.GetPreviousBlockTime` with go conventions by returning the standard `time.Time{}`.

**Review by rhuairahrighairidh — APPROVED** [review]
looks good, liking the increased coverage

**Comment by rhuairahrighairidh** [comment]
nit: do you want to remove `kavadist/doc.go`? We used to use these but switched to `spec/` folders which get pulled into our autogenerated docs.

**Comment by rhuairahrighairidh** [comment]
Something else I ran into: we'll need to hook up the kavadist begin blocker to the app. It's on lines https://github.com/Kava-Labs/kava/blob/cecb3cb019b25b8a0ca61e3ac7313d559638c3b5/app/app.go#L328-L334.

**Pull request merged**